lib/spack/docs/testing_guide.rst

```diff
@@ -0,0 +1,126 @@
+.. _testing-guide:
+
+=============
+Testing Guide
+=============
+
+This guide is intended for developers or administrators who want to
+integrate or test software releases built with Spack.
+
+---------
+Reporting
+---------
+The install and testing infrastructure of Spack supports multiple output formats for
+reporting the result of package builds in tools such as Jenkins or CDash.
+
+The output formats can be enabled by e.g.
+
+.. code-block:: console
+
+   $ spack install --log-format=junit <spec>
+
+Per default the log results will be placed into `var/<format>/test-<short_spec>.xml`.
+To change the default behaviour, use
+
+.. code-block:: console
+
+   $ spack install --log-file report --log-format=junit <spec>
+
+The logs contain one test case per package being built.
+
+
+
+----------------
+using test-suite
+----------------
+
+Test-suite is designed to read in a yaml file describing all packages and compilers along 
+with versions. Using the enabled field will allow you to focus on specific packages.
+To narrow down the scope of a package or compiler, you can use the exclusion field.
+
+--Example of a yaml file---
+
+enable: [bzip2, libelf, .. ,libdwarf]
+
+exclusions: []
+
+packages:
+  - abinit:
+    - versions: [8.0.8b]
+  - ack:
+    - versions: [2.14]
+
+  compilers:
+  - gcc:
+    - versions: [4.9.0, ... 4.7.1, 4.6.3, 4.6.1]
+  - clang:
+    - versions: [7.3.0, 3.4, ... 3.1]
+  
+dashboard: ["https://spack.io/cdash/submit.php?project=spack"]
+
+path: "~/home/username"
```

Comment:
Output path should be a parameter of the command -- not a hard-coded value in the file.  I think it should be possible to either specify an existing output directory on the command line, or to run without `output` and just spit the results out to a new directory *within* the current working dir, named, e.g., `spack-tests-YYYY-MM-DD-timestamp`.  See `spack mirror create` for similar semantics.